addons/stock/migrations/9.0.1.1/post-migration.py

```diff
@@ -50,15 +52,35 @@ def _migrate_pack_operation(env):
         {'processed': AsIs(openupgrade.get_legacy_name('processed'))})
 
 
-def _migrate_stock_picking(cr):
-    cr.execute(
-        "update stock_picking p set "
-        "location_id=coalesce(location_id, default_location_src_id), "
-        "location_dest_id=coalesce(location_dest_id, "
-        "default_location_dest_id) "
-        "from stock_picking_type t "
-        "where p.picking_type_id=t.id and "
-        "(location_id is null or location_dest_id is null)")
+def _migrate_stock_picking(env):
+
+    env.cr.execute(
+        "update stock_picking set "
+        "recompute_pack_op = False "
+        "where state in ('done', 'cancel')")
+
+    env.cr.execute("""
+        UPDATE stock_picking sp
+        SET location_id = sm.location_id,
+        location_dest_id = sm.location_dest_id
+        FROM stock_move sm
+        WHERE sm.picking_id = sp.id
+    """)
+
+    env.cr.execute("""
+        UPDATE stock_move sm
+        SET location_id = sp.location_id,
+        location_dest_id = sp.location_dest_id
+        FROM stock_picking sp
+        WHERE sp.id = sm.picking_id
+        AND sm.state NOT IN ('done', 'cancel')
+    """)
```

Comment:
It looks like you copy the locations from the stock moves to the pickings, and then back again from the pickings to the stock moves. Am I missing something?